backend/src/game_server.rs

```diff
@@ -36,6 +36,57 @@ impl GameServer {
     fn check_player_turn(&self, addr: &Addr<Websocket>) -> bool {
         self.clients[self.current_player] == *addr
     }
+
+    fn check_action_available(&mut self, addr: &Addr<Websocket>, action: Action) -> bool {
+       // if it's not player's turn no actions are availble
+       if self.check_player_turn(addr) == false {
+           return false
+       }
+
+       match action {
+           Action::Select | Action::Deselect => {
+               // if player does not have any selections, they can select and deselect
+               let hex = match self.game.selected_hex {
+                   Some(hex) => hex,
+                   None => return true,
+               };
+
+               let unit = match hex.get_unit() {
+                   Some(unit) => unit,
+                   None => return false,
```

Comment:
should be error

Reply:
Repeated block with assigning hex and unit variables.
Better to move above.

Reply:
you will have to check if they none or some even if you get them above

Reply:
not sure should the error be here: need to discuss